lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -71,7 +70,14 @@ def serializable_hash(options = nil)
 
         def fragment_cache(cached_hash, non_cached_hash)
           root = false if instance_options.include?(:include)
-          ActiveModel::Serializer::Adapter::JsonApi::FragmentCache.new.fragment_cache(root, cached_hash, non_cached_hash)
+          core_cached       = cached_hash.first
+          core_non_cached   = non_cached_hash.first
+          no_root_cache     = cached_hash.delete_if { |key, _value| key == core_cached[0] }
+          no_root_non_cache = non_cached_hash.delete_if { |key, _value| key == core_non_cached[0] }
+          cached_resource   = (core_cached[1]) ? core_cached[1].deep_merge(core_non_cached[1]) : core_non_cached[1]
+          hash = (root) ? { root => cached_resource } : cached_resource
```

Comment:
I know this code is pre-existing but could you give an explanation of what's going on here. I can't seem to understand what this block of code is supposed to do. I especially don't get why we would have a `root` here.